go/store/nbs/store.go

```diff
@@ -273,12 +272,13 @@ func (nbs *NomsBlockStore) conjoinIfRequired(ctx context.Context) (bool, error)
 func (nbs *NomsBlockStore) UpdateManifest(ctx context.Context, updates map[hash.Hash]uint32) (mi ManifestInfo, err error) {
 	err = nbs.checkAllManifestUpdatesExist(ctx, updates)
 	if err != nil {
-		return
+		return manifestContents{}, err
 	}
-	return nbs.updateManifestAddFiles(ctx, updates, nil)
+	mi, _, err = nbs.updateManifestAddFiles(ctx, updates, nil, nil, nil)
+	return mi, err
 }
 
-func (nbs *NomsBlockStore) updateManifestAddFiles(ctx context.Context, updates map[hash.Hash]uint32, appendixOption *ManifestAppendixOption) (mi ManifestInfo, err error) {
+func (nbs *NomsBlockStore) updateManifestAddFiles(ctx context.Context, updates map[hash.Hash]uint32, appendixOption *ManifestAppendixOption, gcGen *hash.Hash, sources chunkSourceSet) (mi ManifestInfo, gcGenDifferent bool, err error) {
```

Comment:
Is it worth defining a type alias for gcGen? Probably useful to distinguish from other random hashes